cmd/cluster-displayer/main.go

```diff
@@ -0,0 +1,199 @@
+)
+
+type Page struct {
+	Data []map[string]string `json:"data"`
+}
+
+func gatherOptions() (options, error) {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.logLevel, "log-level", "info", "Level at which to log output.")
+	fs.IntVar(&o.port, "port", 8081, "Port to run the server on")
+	fs.StringVar(&o.hiveKubeconfigPath, "hive-kubeconfig", "", "Path to the kubeconfig file to use for requests to Hive.")
+	fs.DurationVar(&o.gracePeriod, "gracePeriod", time.Second*10, "Grace period for server shutdown")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		return o, fmt.Errorf("failed to parse flags: %w", err)
+	}
+	return o, nil
+}
+
+func validateOptions(o options) error {
+	_, err := logrus.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid --log-level: %w", err)
+	}
+	if o.hiveKubeconfigPath == "" {
+		return fmt.Errorf("--hive-kubeconfig must be set")
+	}
+	return nil
+}
+
+type options struct {
+	logLevel           string
+	port               int
+	gracePeriod        time.Duration
+	hiveKubeconfigPath string
+}
+
+func addSchemes() error {
+	if err := hivev1.AddToScheme(scheme.Scheme); err != nil {
+		return fmt.Errorf("failed to add hivev1 to scheme: %w", err)
+	}
+	return nil
+}
+
+func getLabelsString(l map[string]string) string {
+	keys := make([]string, 0, len(l))
+	for k := range l {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	var s []string
+	for _, k := range keys {
+		s = append(s, fmt.Sprintf("%s=%s", k, l[k]))
+	}
+	strings.Join(s, ",")
+	return strings.Join(s, ",")
+}
+
+func getClusterPoolPage(ctx context.Context, hiveClient ctrlruntimeclient.Client) (*Page, error) {
+	clusterPools := &hivev1.ClusterPoolList{}
+	if err := hiveClient.List(ctx, clusterPools); err != nil {
+		return nil, fmt.Errorf("failed to list cluster pools: %w", err)
+	}
+	var page Page
+	page.Data = []map[string]string{}
+	for _, p := range clusterPools.Items {
+		maxSize := "nil"
+		if p.Spec.MaxSize != nil {
+			maxSize = strconv.FormatInt(int64(*p.Spec.MaxSize), 10)
+		}
+		page.Data = append(
+			page.Data, map[string]string{
+				"namespace": p.Namespace,
+				"name":      p.Name,
+				"ready":     strconv.FormatInt(int64(p.Status.Ready), 10),
+				"size":      strconv.FormatInt(int64(p.Spec.Size), 10),
+				"maxSize":   maxSize,
+				"imageSet":  p.Spec.ImageSetRef.Name,
+				"labels":    getLabelsString(p.Labels),
+			},
+		)
+	}
+	return &page, nil
+}
+
+func getResource(ctx context.Context, hiveClient ctrlruntimeclient.Client, resource string) (*Page, error) {
+	switch strings.ToLower(resource) {
+	case "clusterpools":
+		return getClusterPoolPage(ctx, hiveClient)
+	default:
+		return nil, fmt.Errorf("unknown resource: %s", resource)
+	}
+}
+
+func getRouter(ctx context.Context, hiveClient ctrlruntimeclient.Client) *mux.Router {
+	handler := mux.NewRouter()
+
+	handler.HandleFunc("/api/health", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Just use Prow's health and readiness utility please

Reply:
https://github.com/openshift/ci-tools/compare/65718c42f4ab6d519e1ae480f745f7e027f1497e..e5ca285632aa9b8bb95dfc73e01d17e1f6e2b19d

I changed it back while defining probe on this service.
It does not feel right to (blackbox-)probe another port.
https://github.com/openshift/release/pull/19404/files#diff-727bc2024b05f660a6c5779c3beb78b988c1e769308f5c45821b4ce4e28402c8R36

Like deck, we have for container's probe.
https://github.com/openshift/release/blob/4adaefb73f7faebf494bcf1e83569f87d20d06f3/clusters/app.ci/prow/03_deployment/deck.yaml#L116

When prometheus probes it, it still uses the UI port.
https://github.com/openshift/release/blob/4adaefb73f7faebf494bcf1e83569f87d20d06f3/clusters/app.ci/prow-monitoring/additional-scrape-configs_secret.yaml#L25

Reply:
`func NewHealthOnPort(port int) *Health` does not help either because I need to modify the `healthMux` to call `HandleFunc`. However, it is not public.